File: 은지/최대공약수와 최소공배수.py
def solution(n, m):
    minV = min(n, m)
    maxV = max(n, m)
    for x in range(1, maxV*minV+1):
        # 최대공약수
        if minV % x == 0 and maxV % x == 0:
            g = x
        # 최소공배수
        if x % minV == 0 and x % maxV == 0:
            l = x
            break
    answer = [g, l]
    return answer

n = 3
m = 12
print(solution(n, m))

File: 은지/최대공약수와 최소공배수.js
function solution(n, m) {
  const min_v = Math.min(n, m)
  const max_v = Math.max(n, m)
  for (let x=0; x<=max_v*min_v; x++) {
    // 최대공약수
    if (min_v % x === 0 && max_v % x === 0) {
      var g = x
    }
    // 최소공배수
    if (x % min_v === 0 && x % max_v === 0) {
      var l = x
    }
    // 최소공배수 구해지면 멈춰!
    if (l) {
      break
    }
  }
  const answer = [g, l]
  return answer;
}